### ISSUE TITLE
chownr calling readdir on network-share file causes ENOTSUP, but isn't handled correctly

### ISSUE THREAD

**Issue by rocketmonkeys** [isaacs/chownr#5]
Same problem as this:
https://github.com/isaacs/node-glob/issues/205

But in chrownr (https://github.com/isaacs/chownr/blob/master/chownr.js#L11), we have:

```
fs.readdir(p, function (er, children) {
    // any error other than ENOTDIR means it's not readable, or
    // doesn't exist.    give up.
    if (er && er.code !== "ENOTDIR") return cb(er)
```

When readdir()'ing a local file, we get ENOTDIR (good).  Same thing on a network path gives ENOTSUP (weird).

We need to support ENOTSUP as well.  Simple fix should be:

```
fs.readdir(p, function (er, children) {
    // any error other than ENOTDIR/ENOTSUP means it's not readable, or
    // doesn't exist.    give up.
    if (er && er.code !== "ENOTDIR" && er.code !== "ENOTSUP") return cb(er)
```

I'll make a small PR for this.

**Comment by isaacs** [comment]
fixed on 03eb97e789bb7496129415af4b77e25d43ab10a7

**Comment by rocketmonkeys** [comment]
Thanks!
